elixir/apps/domain/lib/domain/changes/hooks/actors.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Domain.Changes.Hooks.Actors do
+  @behaviour Domain.Changes.Hooks
+  alias __MODULE__.DB
+
+  @impl true
+  def on_insert(_lsn, _data), do: :ok
+
+  @impl true
+
+  # Delete api_tokens, client_tokens, and portal_sessions when an actor is disabled
+  def on_update(
+        _lsn,
+        %{
+          "disabled_at" => nil,
+          "account_id" => account_id,
+          "id" => actor_id
+        },
+        %{"disabled_at" => disabled_at}
+      )
+      when not is_nil(disabled_at) do
+    DB.delete_api_tokens_for_actor(account_id, actor_id)
```

Comment:
Since API tokens are long lived, should we deleted them or just disable them?

Reply:
Hmmm. I could see it either way, but keeping them is probably going to result in fewer surprises.

TBH I think this is where our domain model breaks down a bit. IMO api tokens are not actors, they are adjacent.

So a subject would either be an actor subject or an API subject to match.

Reply:
We can save that for another refactor.

Reply:
Or, API tokens _are_ actors, but we change the relationship such that an API actor `has_one` API token.